csrc/device_lower/analysis/circular_buffer.cpp

```diff
@@ -769,4 +770,66 @@ IterDomain* getCircularBufferAxis(const TensorView* tv) {
   return tv->axis(cb_axis);
 }
 
+std::vector<AsyncWarp> createAsyncWarps(const std::vector<Expr*>& exprs) {
+  std::vector<AsyncWarp> async_warps;
+
+  // Gather all async operations.
+  // TODO Add support for blackwell MmaOp
+  std::vector<Expr*> async_warp_exprs;
+  std::copy_if(
+      exprs.begin(),
+      exprs.end(),
+      std::back_inserter(async_warp_exprs),
+      [](Expr* e) { return ir_utils::isCpAsyncBulkLoad(e); });
+
+  // short-circuit: No async operations detected.
+  if (async_warp_exprs.size() <= 1) {
+    return async_warps;
+  }
+
+  // TODO Divide into operations into separate AsyncWarps for multi-role
+  // specialization. The current assumption is a single AsyncWarp with same
+  // stage_slice_position.
+
+  // Get TensorViews for async warps.
+  std::vector<TensorView*> async_warp_tvs;
+  std::transform(
+      async_warp_exprs.begin(),
+      async_warp_exprs.end(),
+      std::back_inserter(async_warp_tvs),
+      [](Expr* e) {
+        auto output_tvs =
+            ir_utils::filterByType<TensorView>(e->outputs()).vector();
+        NVF_ERROR(output_tvs.size() == 1);
+        return output_tvs.front();
+      });
+  NVF_ERROR(async_warp_tvs.size() > 1);
+
+  // Check that all operations in the same warp have the same
+  // stage_slice_position.
+  std::vector<int64_t> stage_slice_positions;
+  std::transform(
+      async_warp_tvs.begin(),
+      async_warp_tvs.end(),
+      std::back_inserter(stage_slice_positions),
+      [](TensorView* tv) {
+        std::optional<int64_t> opt_stage_slice_position =
+            ir_utils::getStageSlicePosition(tv);
+        return opt_stage_slice_position.value_or(-1);
+      });
+  NVF_ERROR(stage_slice_positions.size() > 1);
+  NVF_ERROR(std::all_of(
+      stage_slice_positions.begin() + 1,
+      stage_slice_positions.end(),
+      [&](int64_t v) { return v == stage_slice_positions.front(); }));
+
+  TensorView* async_warp_tv = async_warp_tvs.front();
+  NVF_ERROR(async_warp_tv != nullptr);
+  int64_t stage_slice_position = stage_slice_positions.front();
+
+  async_warps.emplace_back(
```

Comment:
Can this compile?
```suggestion
async_warps.push_back(AsyncWarp{async_warp_exprs, async_warp_tvs, stage_slice_position});
```

Reply:
It compiles https://godbolt.org/z/TfKMhPseG.

> Appends a new element to the end of the container. The element is constructed through [std::allocator_traits::construct](https://en.cppreference.com/w/cpp/memory/allocator_traits/construct.html), which typically uses placement new to construct the element in-place at the location provided by the container. The arguments args... are forwarded to the constructor as [std::forward](https://en.cppreference.com/w/cpp/utility/forward.html)<Args>(args)....

from https://en.cppreference.com/w/cpp/container/vector/emplace_back

Why shouldn't it compile?

Reply:
I thought universal initialization using `{}` is required, but looks like not the case.